scalableminds/webknossos
# Make segmentation output layer name for neuron detection configurable

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- I suggest testing locally
- In the `applications.conf` configure `isWkorgInstance=true` and `jobsEnabled=true`
- Open an annotation of a dataset with at least one color layer.
- Open the jobs modal via the "AI Analysis" button in the toolbar
- The default and only available neuron detection job should now have the field to configure the segmentation layer name available.
- Configure an arbitrary valid name (existing layer names should not be allowed)
- Open the network tab of the browser console
- Start the job and inspect whether the correct output segmentation layer name has been sent.

### TODOs:
- [x] Investigate, whether the disallowed layer names should include the name of all color layers (which is afaik currently not the case)
  - The layer names need to be unique.
- [x] Check which layers are copied over to the new data set by the neuron detection job.
  - Code reads like all layers are copied over and the prediction is added as a new segmentation layer
  - [x] -> Do not allow any existing layer name in the start jobs modal
- [x] Configure neuron detection job to use new segmentation layer name.

### Issues:
- fixes [#6197](https://github.com/scalableminds/webknossos/issues/6197)

------
(Please delete unneeded items, merge only when none are left open)
